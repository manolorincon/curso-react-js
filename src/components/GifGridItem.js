import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ id, title, url }) => {

    let widthImg;

    return ( 
        <div className="card">
            <img 
                src={ url } 
                id={id} 
                alt={ title }
                ref={
                    (input) => {                        
                        if (!input) { return; }
                        const img = input;
                        widthImg = img.width;
                    }
                }
            >
            </img>
            <p style={{ maxWidth: `${widthImg}px` }}> { title }</p>
        </div>
    );
}

GifGridItem.propTypes = {
    url:PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
 
export default GifGridItem;