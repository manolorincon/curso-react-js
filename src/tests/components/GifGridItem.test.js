import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import {shallow} from 'enzyme';

describe('Primer test para componentes', () => {

    test('Debe retornar OK', () => {

        const title = 'Goku'

        const url = 'https://static.t13.cl/images/original/2019/05/1557689182-goku.jpg'

        const wrapper = shallow(<GifGridItem title = {title} url = {url} />)
        
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
