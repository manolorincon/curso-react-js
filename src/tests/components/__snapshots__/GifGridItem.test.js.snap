// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primer test para componentes Debe retornar OK 1`] = `
<div
  className="card"
>
  <img
    alt="Goku"
    src="https://static.t13.cl/images/original/2019/05/1557689182-goku.jpg"
  />
  <p
    style={
      Object {
        "maxWidth": "undefinedpx",
      }
    }
  >
     
    Goku
  </p>
</div>
`;
